Use the logger object for outputting debug information

whoward opened the pull request:
This can be pretty distracting if you're running the full suite and just want to know your progress (we're using rspec with the documentation formatter)

This change will not modify anything from the average user's perspective but lets others squelch the output using a line like this:

```ruby
Appium::Logger.level = Logger::WARN
```

cyrilrbt commented:
:+1:

mclark commented:
:+1:

bootstraponline commented:
Thanks for the pull request! I think using a logger is a great idea. Note that some methods exist only to output to stdout (intended to be used from the pry console). I think in those situations, replacing puts doesn't make sense.

> This change will not modify anything from the average user's perspective but lets others squelch the output using a line like this:
> 
> Appium::Logger.level = Logger::WARN

I think this should be the default.

bootstraponline commented on lib/appium_lib/android/helper.rb:
```diff
@@ -152,7 +152,7 @@ def get_android_inspect class_name=false
     def page opts={}
       class_name = opts.is_a?(Hash) ? opts.fetch(:class, nil) : opts
-      puts get_android_inspect class_name
+      Appium::Logger.info get_android_inspect class_name
```
This exists only to output to stdout, if you want to use it without messages then invoke get_android_inspect.

whoward replied:
true, I haven't used these methods before - they actually look kind of useful

bootstraponline commented on lib/appium_lib/common/helper.rb:
```diff
@@ -108,7 +108,7 @@ def get_page_class
     # Useful for appium_console.
     def page_class
-      puts get_page_class
+      Appium::Logger.info get_page_class
```
This exists only to output to stdout, if you want to use it without messages then invoke get_page_class.

bootstraponline commented on lib/appium_lib/common/patch.rb:
```diff
@@ -90,7 +90,7 @@ def raw_execute(command, opts = {}, command_hash = nil)
       path_str   = path.sub(path_match[0], '') unless path_match.nil?
 
-      puts "#{verb} #{path_str}"
+      Appium::Logger.info  "#{verb} #{path_str}"
```
This function does nothing other than print to stdout so I don't think a logger makes sense here.

bootstraponline commented on lib/appium_lib/driver.rb:
```diff
@@ -319,9 +319,9 @@ def initialize opts={}
-        ap opts unless opts.empty?
-        puts "Debug is: #{@appium_debug}"
-        puts "Device is: #{@appium_device}"
+        Appium::Logger.ap opts unless opts.empty?
```
This should probably remove the `@appium_debug` construct and then invoke debug on the logger.

bootstraponline commented on lib/appium_lib/ios/helper.rb:
```diff
@@ -55,25 +55,25 @@ def fix_space s
-            puts "#{type}" if name || label || value || hint
-            puts "   name, label, value: #{name}" if name
-            puts "   hint: #{hint}" if hint
+            Appium::Logger.info "#{type}" if name || label || value || hint
```
This function is meant to output to stdout so using the logger here doesn't make sense.

whoward commented:
okay back to you - i reverted a the puts that actually should be there, updated the log scope to use debug in that @appium_debug section and also set the default log level to WARN

I didn't end up removing the @appium_debug block because of the call to `patch_webdriver_bridge` - this isn't something you can easily undo if you at some point in the program execution you decide you want to increase or decrease your logger level.  I have some ideas on how that could be achieved but its probably best left for another PR.

bootstraponline commented:
Ok, makes sense. Please squash to one commit then I'll merge.

```
git reset --soft head~4
git commit -am "Use the logger object for outputting debug information"
git push -f
```

whoward commented:
@bootstraponline done! back to you.

bootstraponline commented:
Thanks! Merged & released to RubyGems as v6.
